Update the shape function syntax

smfr opened the pull request:
#### 3b0dff1a6be8105e2ceb0c5757d3f0e806af5798
<pre>
Update the shape function syntax
<a href="https://bugs.webkit.org/show_bug.cgi?id=281674">https://bugs.webkit.org/show_bug.cgi?id=281674</a>
<a href="https://rdar.apple.com/138126105">rdar://138126105</a>

Reviewed by Antti Koivisto.

This PR implements the following changes to the `shape()` syntax, proposed at [1]:
- Absolute coordinate values can now take `&lt;position&gt;` values.
- `from start`/`from end`/`from origin` are now supported on control points
- `using` is changed to `with`
- Multiple control points are separated with `/`

To support the start/end/origin-relative controls points, add `ControlPointAnchoring` and classes
for the CSSValue (`ControlPointValue`) and style (`ControlPoint`) data structures. BasicShapesShape
gains some logic to compute the control point positions based on the relevant point.

BasicShapeConversion.cpp now exports some `toLength` etc helpers to avoid code duplication.

The impact of start/end/origin-relative controls points on interpolation matches my proposal at [2].

[1] <a href="https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2412816773">https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2412816773</a>
[2] <a href="https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2418266716">https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2418266716</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape-control-points.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape-control-points.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-path-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-path-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-005-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-005.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-006-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-006.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-007.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-007.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-008.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-008.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-009.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-009.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-010.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-010.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-path-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-shape-control-points-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/parsing/clip-path-shape-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/parsing/clip-path-shape-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-computed.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-computed.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-invalid.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-invalid.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-valid.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-valid.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-path-interpolation-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-path-interpolation-008.html:
* Source/WebCore/css/BasicShapeConversion.cpp:
(WebCore::convertToLengthOrAuto):
(WebCore::convertToLength):
(WebCore::convertToLengthSize):
(WebCore::coordinatePairToLengthPoint):
(WebCore::positionOrCoordinatePairToLengthPoint):
(WebCore::basicShapeShapeForValue):
(WebCore::convertToLengthPoint): Deleted.
* Source/WebCore/css/BasicShapeConversion.h:
* Source/WebCore/css/BasicShapesShapeSegmentConversion.cpp:
(WebCore::controlPointValue):
(WebCore::toCSSShapeSegmentValue):
* Source/WebCore/css/CSSBasicShapes.cpp:
(WebCore::CSSShapeValue::create):
(WebCore::CSSShapeValue::CSSShapeValue):
* Source/WebCore/css/CSSBasicShapes.h:
* Source/WebCore/css/CSSShapeSegmentValue.cpp:
(WebCore::controlPoint):
(WebCore::ControlPointValue::cssText const):
(WebCore::CSSShapeSegmentValue::createCubicCurve):
(WebCore::CSSShapeSegmentValue::createQuadraticCurve):
(WebCore::CSSShapeSegmentValue::createSmoothCubicCurve):
(WebCore::CSSShapeSegmentValue::customCSSText const):
(WebCore::CSSShapeSegmentValue::toShapeSegment const):
* Source/WebCore/css/CSSShapeSegmentValue.h:
(WebCore::ControlPointValue::operator== const):
(WebCore::CSSShapeSegmentValue::OnePointData::OnePointData):
(WebCore::CSSShapeSegmentValue::TwoPointData::TwoPointData):
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeShapeToCoordinates):
(WebCore::CSSPropertyParserHelpers::consumeShapeRelativeControlPoint):
(WebCore::CSSPropertyParserHelpers::consumeShapeCommand):
(WebCore::CSSPropertyParserHelpers::consumeBasicShapeShape):
(WebCore::CSSPropertyParserHelpers::consumeCoordinatePair): Deleted.
* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/BasicShapes.h:
* Source/WebCore/rendering/style/BasicShapesShape.cpp:
(WebCore::BasicShapeShape::canBlend):
(WebCore::BasicShapeShape::blend):
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/BasicShapesShape.h:

Canonical link: <a href="https://commits.webkit.org/285442@main">https://commits.webkit.org/285442@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cca4a526c360480f607624e18462cfc081bce4b9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72608 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52033 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25406 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76796 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/23849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/74723 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59838 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23642 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57140 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/23849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75675 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47086 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62522 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/37567 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43736 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19989 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22171 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65596 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20348 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78467 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16854 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19485 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65588 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16902 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64857 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16046 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13151 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6802 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47831 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/2618 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48898 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/50193 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48643 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on Source/WebCore/css/BasicShapesShapeSegmentConversion.cpp:
```diff
@@ -54,6 +54,14 @@ static auto lengthSizeToCSSValue(const LengthSize& value, const RenderStyle& sty
         CSSPrimitiveValue::create(value.height, style));
 }
 
+static auto controlPointToControlPointValue(const ControlPoint& controlPoint, const RenderStyle& style)
```
We don't use `auto` for return type for this case generally

darinadler replied:
Buy _why_ don’t we?

I think we can just name this function `controlPointValue` and while others rarely choose this style, I might even name it just `value` since it’s local and unambiguous due to the argument type. Don’t need to say the name of an argument type, and the shorter name feels easier to read.

mdubet commented on Source/WebCore/rendering/style/BasicShapesShape.cpp:
```diff
@@ -375,6 +416,8 @@ Path BasicShapeShape::path(const FloatRect& referenceRect) const
     SVGPathParser::parse(pathSource, builder);
     path.translate(toFloatSize(referenceRect.location()));
+
+    WTF_ALWAYS_LOG("BasicShapeShape::path " << path);
```
Should log to a channel

smfr replied:
Left in by mistake.

darinadler commented on Source/WebCore/css/BasicShapeConversion.h:
```diff
@@ -49,8 +49,19 @@ class CSSToLengthConversionData;
+struct LengthPoint;
+struct LengthSize;
+
+enum class CoordinateAffinity : uint8_t;
```
In other headers I’ve been sorting alphabetically so `enum class` goes between `class` and `struct`.

darinadler commented on Source/WebCore/css/CSSShapeSegmentValue.cpp:
```diff
@@ -27,17 +27,44 @@
 
 namespace WebCore {
 
+static ControlPoint controlPointValueFromValue(const ControlPointValue& controlPointValue, CoordinateAffinity affinity, const Style::BuilderState& builderState)
```
Just name this `controlPoint`?

darinadler commented on Source/WebCore/rendering/style/BasicShapes.cpp:
```diff
@@ -846,6 +846,16 @@ TextStream& operator<<(TextStream& ts, CoordinateAffinity affinity)
+TextStream& operator<<(TextStream& ts, ControlPointAnchoring anchoring)
+{
+    switch (anchoring) {
+    case ControlPointAnchoring::FromStart: ts << "from start"_s; break;
+    case ControlPointAnchoring::FromEnd: ts << "from end"_s; break;
+    case ControlPointAnchoring::FromOrigin: ts << "from origin"_s; break;
```
Could do `return ts << ...` here.

darinadler commented on Source/WebCore/rendering/style/BasicShapes.cpp:
```diff
@@ -846,6 +846,16 @@ TextStream& operator<<(TextStream& ts, CoordinateAffinity affinity)
+    case ControlPointAnchoring::FromEnd: ts << "from end"_s; break;
+    case ControlPointAnchoring::FromOrigin: ts << "from origin"_s; break;
+    }
+    return ts;
```
Then could `ASSERT_NOT_REACHED()` here.

darinadler commented on Source/WebCore/rendering/style/BasicShapes.h:
```diff
@@ -496,6 +502,7 @@ class BasicShapeXywh final : public BasicShape {
 };
 
 WTF::TextStream& operator<<(WTF::TextStream&, CoordinateAffinity);
+WTF::TextStream& operator<<(WTF::TextStream&, ControlPointAnchoring);
```
Alphabetical order would reverse these two.

darinadler commented on Source/WebCore/rendering/style/BasicShapesShape.cpp:
```diff
@@ -244,6 +280,11 @@ class ShapeConversionPathConsumer final : public SVGPathConsumer {
         return { Length(p.x(), LengthType::Fixed), Length(p.y(), LengthType::Fixed) };
     }
 
+    static ControlPoint controlPointFromPoint(FloatPoint p)
```
Could just name this `controlPoint`.

darinadler commented on Source/WebCore/rendering/style/BasicShapesShape.cpp:
```diff
@@ -244,6 +280,11 @@ class ShapeConversionPathConsumer final : public SVGPathConsumer {
 
+    static ControlPoint controlPointFromPoint(FloatPoint p)
+    {
+        return ControlPoint { fromPoint(p), { } };
```
Might not need to do `ControlPoint { }` here, just `{ }` may work.

smfr committed:
Update the shape function syntax
https://bugs.webkit.org/show_bug.cgi?id=281674
rdar://138126105

Reviewed by Antti Koivisto.

This PR implements the following changes to the `shape()` syntax, proposed at [1]:
- Absolute coordinate values can now take `<position>` values.
- `from start`/`from end`/`from origin` are now supported on control points
- `using` is changed to `with`
- Multiple control points are separated with `/`

To support the start/end/origin-relative controls points, add `ControlPointAnchoring` and classes
for the CSSValue (`ControlPointValue`) and style (`ControlPoint`) data structures. BasicShapesShape
gains some logic to compute the control point positions based on the relevant point.

BasicShapeConversion.cpp now exports some `toLength` etc helpers to avoid code duplication.

The impact of start/end/origin-relative controls points on interpolation matches my proposal at [2].

[1] https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2412816773
[2] https://github.com/w3c/csswg-drafts/issues/10649#issuecomment-2418266716

* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape-control-points.tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape-control-points.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/animations/clip-path-interpolation-shape.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-path-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-path-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-001-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-005-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-005.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-006-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-006.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-007.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-007.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-008.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-008.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-009.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-009.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-010.tentative-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/clip-path-shape-010.tentative.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-path-001-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/clip-path/reference/clip-path-shape-control-points-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/parsing/clip-path-shape-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-masking/parsing/clip-path-shape-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-computed.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-computed.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-invalid.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-invalid.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-valid.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-shapes/shape-functions/shape-function-valid.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-path-interpolation-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-path-interpolation-008.html:
* Source/WebCore/css/BasicShapeConversion.cpp:
(WebCore::convertToLengthOrAuto):
(WebCore::convertToLength):
(WebCore::convertToLengthSize):
(WebCore::coordinatePairToLengthPoint):
(WebCore::positionOrCoordinatePairToLengthPoint):
(WebCore::basicShapeShapeForValue):
(WebCore::convertToLengthPoint): Deleted.
* Source/WebCore/css/BasicShapeConversion.h:
* Source/WebCore/css/BasicShapesShapeSegmentConversion.cpp:
(WebCore::controlPointValue):
(WebCore::toCSSShapeSegmentValue):
* Source/WebCore/css/CSSBasicShapes.cpp:
(WebCore::CSSShapeValue::create):
(WebCore::CSSShapeValue::CSSShapeValue):
* Source/WebCore/css/CSSBasicShapes.h:
* Source/WebCore/css/CSSShapeSegmentValue.cpp:
(WebCore::controlPoint):
(WebCore::ControlPointValue::cssText const):
(WebCore::CSSShapeSegmentValue::createCubicCurve):
(WebCore::CSSShapeSegmentValue::createQuadraticCurve):
(WebCore::CSSShapeSegmentValue::createSmoothCubicCurve):
(WebCore::CSSShapeSegmentValue::customCSSText const):
(WebCore::CSSShapeSegmentValue::toShapeSegment const):
* Source/WebCore/css/CSSShapeSegmentValue.h:
(WebCore::ControlPointValue::operator== const):
(WebCore::CSSShapeSegmentValue::OnePointData::OnePointData):
(WebCore::CSSShapeSegmentValue::TwoPointData::TwoPointData):
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeShapeToCoordinates):
(WebCore::CSSPropertyParserHelpers::consumeShapeRelativeControlPoint):
(WebCore::CSSPropertyParserHelpers::consumeShapeCommand):
(WebCore::CSSPropertyParserHelpers::consumeBasicShapeShape):
(WebCore::CSSPropertyParserHelpers::consumeCoordinatePair): Deleted.
* Source/WebCore/rendering/style/BasicShapes.cpp:
(WebCore::operator<<):
* Source/WebCore/rendering/style/BasicShapes.h:
* Source/WebCore/rendering/style/BasicShapesShape.cpp:
(WebCore::BasicShapeShape::canBlend):
(WebCore::BasicShapeShape::blend):
(WebCore::operator<<):
* Source/WebCore/rendering/style/BasicShapesShape.h:

Canonical link: https://commits.webkit.org/285442@main
